pkg/controller/attach_detach.go

```diff
@@ -573,19 +551,27 @@ func reportError(azVolumeAttachment *azdiskv1beta2.AzVolumeAttachment, state azd
 	return updateState(azVolumeAttachment, state, forceUpdate)
 }
 
-func (r *ReconcileAttachDetach) recreateAzVolumeAttachment(ctx context.Context, syncedVolumeAttachments map[string]bool, volumesToSync map[string]bool) (map[string]bool, map[string]bool, error) {
+func (r *ReconcileAttachDetach) recreateAzVolumeAttachmentWithPrimaryRole(
+	ctx context.Context,
+	syncedVolumeAttachments map[string]bool,
+	volumeHandles map[string]string,
+	volumeAttributes map[string]map[string]string) (map[string]bool, map[string]string, map[string]map[string]string, error) {
```

Comment:
Instead of passing and returning `map[string]volumeInfo`, consider adding this as a field of `ReconcileAttachDetach`. This would clean this up a bit and make it more efficient.

This would also enable us to track synced state of the volume by adding a `synced bool` field to the `volumeInfo` struct instead of maintaining a separate map.

Reply:
Done.